src/content/en/fundamentals/accessibility/01-focus/focus-eight.md

```diff
@@ -0,0 +1,43 @@
+---
+layout: shared/narrow
+title: "Modals and Keyboard Traps"
+description: "Managing focus with modal elements and keyboard traps"
+published_on: 2016-03-01
+updated_on: 2016-03-01
+order: 8
+translation_priority: 0
+authors:
+  - megginkearney
+  - dgash
+key-takeaways:
+  tldr: 
+    - "Learn what accessibility means and how it applies to web development."
+    - "Learn how to make web sites accessible and usable for everyone."
+    - "Learn how to include basic accessibility with minimal development impace."
+    - "Learn what HTML features are available and how to use them to improve accessibility."
+    - "Learn about advanced accessibility techniques for creating polished accessibility experiences."
+notes:
+  efficiency:
+    - "Understanding the accessibility issue, its scope, and its impact can make you a better web developer."
+  problem-solving:
+    - "Catching, identifying, and removing potential accessibility roadblocks before they happen can improve your development process and reduce maintenance requirements."
+---
+
+Sometimes when you're managing focus you can get into a situation you can't get out of. Consider an autocomplete widget that tries to manage focus and captures the tab behavior, but prevents the user from leaving it until it's complete. This is called a *keyboard trap*, and it can be very frustrating for the user. Section 2.1.2 of the Web AIM checklist addresses this issue, stating that keyboard focus should never be locked or trapped at one particular page element. The user should be able to navigate to and from all page elements using only the keyboard.
+
+Oddly, there are times when this behavior is actually desirable, like in a modal window. Normally, when the modal is displayed, you don't want the user to access the content behind it. You might add an overlay to cover the page visually, but that doesn't stop keyboard focus from accidentally traveling outside the modal. In instances like this you can implement a temporary keyboard trap to ensure that you trap focus only while the modal is displayed and then restore focus to the previously-focused item when the modal is closed.
+
+>There are some proposals on how to make this easier for developers, including the `<dialog>` element, but they don't yet have widespread browser support.
+
+Consider a modal dialog represented by a `div` that contains a few elements, and another `div` that represents a background overlay. Let's walk through the basic steps needed to implement a temporary keyboard trap in this situation.
```

Comment:
Would be good to work this link in there https://accessibility.oit.ncsu.edu/training/aria/modal-window/version-3/